src/python/circllhist/__init__.py

```diff
@@ -0,0 +1,77 @@
+import circllhist.ffi as ffi
+
+
+class circllbin:
+    def __init__(self, b):
+        self.b = b
+
+    @classmethod
+    def new(cls):
+        return cls(ffi.ffi.new("hist_bucket_t*"))
+
+    @classmethod
+    def from_number(cls, n):
+        return cls(ffi.C.double_to_hist_bucket(n))
+
+    def __str__(self):
+        buf = ffi.ffi.new("char[]",9)
+        ffi.C.hist_bucket_to_string(self.b, buf)
+        return str(ffi.ffi.string(buf), "ASCII")
+
+    def width(self):
+        return ffi.C.hist_bucket_to_double_bin_width(self.b)
+
+    def midpoint(self):
+        return ffi.C.hist_bucket_midpoint(self.b)
+
+    def edge(self):
+        "Returns the edge of the histogram bucket that is closer to zero"
+        return ffi.C.hist_bucket_to_double(self.b)
+
+
+class circllhist:
```

Comment:
Explicitly inherit from `object` to always use "new style classes" for Python 2 and 3 consistency. 

And use `__slots__`.